crates/aiken-lang/src/parser/expr/bytearray.rs

```diff
@@ -0,0 +1,34 @@
+use chumsky::prelude::*;
+
+use crate::{
+    expr::UntypedExpr,
+    parser::{error::ParseError, token::Token, utils},
+};
+
+pub fn parser() -> impl Parser<Token, UntypedExpr, Error = ParseError> {
+    utils::bytearray().map_with_span(|(preferred_format, bytes), span| UntypedExpr::ByteArray {
```

Comment:
Any good reason why `bytearray` is under `utils` and not under `expr/bytearray`? I simply find it odd to find that code there. 

A suggestion: rename and move `utils/bytearray` to `expr/bytearray/bytes`. 

Or possibly, have the bytearray's parser take an argument that indicates whether it should be wrapped as an `UntypedExpr` or as `Constant` (since that the only two places it's used)